src/module.c

```diff
@@ -1375,26 +1375,57 @@ int RM_BlockedClientMeasureTimeEnd(RedisModuleBlockedClient *bc) {
     return REDISMODULE_OK;
 }
 
-/* This API allows modules to let Redis process some commands during long
- * blocking execution of a module command. The module can call this API
- * periodically, and after the time defined by the `script-time-limit` config,
- * Redis will start rejecting most commands with `-BUSY` error, but allow the
- * ones marked with the `allow-busy` flag to be executed. This API can also be
- * used in thread safe context (while locked), and during loading (in the
- * rdb_load, in which case it'll reject commands with -LOADING error) */
-void RM_Yield(RedisModuleCtx *ctx, const char *busy_reply) {
+/* This API allows modules to let Redis process background tasks, and some
+ * commands during long blocking execution of a module command.
+ * The module can call this API periodically.
+ * The flags is a bit mask of these:
+ *
+ * - `REDISMODULE_YIELD_FLAG_NONE`: No special flags, can perform some background
+ *                                  operations, but not process client commands.
+ * - `REDISMODULE_YIELD_FLAG_CLIENTS`: Redis can also process client commands.
+ *
+ * The `busy_reply` argument is optional, and can be used to control the verbose
+ * error string after the `-BUSY` error code.
+ *
+ * When the `REDISMODULE_YIELD_FLAG_CLIENTS` is used, Redis will only start
+ * processing client commands after the time defined by the
+ * `busy-reply-threshold` config, in which case Redis will start rejecting most
+ * commands with `-BUSY` error, but allow the ones marked with the `allow-busy`
+ * flag to be executed.
+ * This API can also be used in thread safe context (while locked), and during
```

Comment:
Does this imply we're OK with this being called from another thread? Sounds like a lot of potential trouble, now or in the future.

Reply:
yes, we intend for this to also be called from a slow background operation that blocks the GIL for long.
there's even a test for it (not that it increases the confidence that it won't be abused or trigger problematic edge cases)

Reply:
I'm really not sure about suddenly having significant "main thread" code flow running from another thread. Even if things don't break, we'll still be running into weird jemalloc thread-local issues, and may regret that with any future work around threading.